system-tests/ft-suite/src/test/kotlin/com/hotels/styx/admin/ProviderAdminInterfaceSpec.kt

```diff
@@ -90,8 +127,106 @@ class ProviderAdminInterfaceSpec : FeatureSpec() {
                         .bodyAs(UTF_8)
                 body shouldContain "/admin/providers/myMonitor/status"
                 body shouldContain "/admin/providers/mySecondMonitor/status"
+                body shouldNotContain "/admin/providers/appA-monitor/status"
+                body shouldNotContain "/admin/providers/appB-monitor/status"
+            }
+        }
+
+        feature("Backcompat healthcheck service endpoints are listed on the Providers admin interface") {
+
+            scenario("Endpoints are registered after server start") {
+                val originsFileContent = """
+                - id: appA
+                  path: "/"
+                  healthCheck:
+                    uri: "http://www/check/me"
+                  origins:
+                  - { id: "appA-01", host: "localhost:${mockServer.port()}" } 
+            """.trimIndent()
+                originsFile.writeText(originsFileContent)
+                styxServer.restart()
+
+                eventually(5.seconds, AssertionError::class.java) {
+                    val body = styxServer.adminRequest("/admin/providers")
+                            .bodyAs(UTF_8)
+                    body shouldContain "/admin/providers/appA-monitor/status"
+                }
+            }
+
+            scenario("The new endpoint returns status information after server restart") {
+                Thread.sleep(5000)
```

Comment:
Why the sleep here? Also 5 seconds is quite long. Is this justified?

Reply:
No, it's not needed at all. It was left over from investigating something earlier in development.